src/client/address_book/impl.h

```diff
@@ -63,6 +63,99 @@ class AddressBookSubscriber;
 /// @brief Address book implementation
 class AddressBook : public AddressBookDefaults {
  public:
+  /// @brief Container for Address Book entry
+  class Entry
+  {
+   public:
+    /// @brief No default copy-constructor
+    Entry(Entry& rhs) = delete;
+    /// @brief No default move-constructor
+    Entry(Entry&& rhs) = delete;
+    /// @brief Create address book entry from a hostname & identity hash
+    /// @param host Kovri hostname
+    /// @param address Kovri identity hash
+    explicit Entry(
+        const std::string&& host,
+        const kovri::core::IdentHash&& address)
```

Comment:
Please, read more about move semantics and other modern C++ feature.
I'd recommend [Effective Modern C++](https://www.amazon.com/Effective-Modern-Specific-Ways-Improve/dp/1491903996)
- Are you sure with [explicit](http://en.cppreference.com/w/cpp/language/explicit) keyword here?
- `const T&&` type doesn't make sense here
- Even so, you're misusing the benefits of move semantics

I would rewrite this ctor the following way:
```cpp
    Entry(
        std::string host,
        kovri::core::IdentHash address)
      : m_Host(std::move(host))
      , m_Address(std::move(address))
    {
      if (host.empty())
      {
         kovri::core::Exception ex;
         ex.Dispatch(__func__);
         throw std::runtime_error("AddressBook: empty entry hostname");
      }
    }
```

Reply:
Just a note here:
if `std::move` is used on the passed `host` value, then `host` should not be used afterwards.

```
if (host.empty()) { ... }
```

Will for sure be true as `host` will not longer have the value you want.

Reply:
Thanks, I fixed the snippet.

Reply:
Thanks for the catch & rewrite. I'm pretty new to move semantics, so this helps a lot.

Had trouble in one of the unit-test cases with `IdentHash` copying, but that may have been from something else.

Will update with the recommended changes. Thanks @btolfa & @brbzull0 :+1: